pkgs/development/tools/rust/cbindgen/default.nix

```diff
@@ -23,6 +24,16 @@ rustPlatform.buildRustPackage rec {
     hash = "sha256-XTGHHD5Qw3mr+lkPKOXyqb0K3sEENW8Sf0n9mtrFFXI=";
   };
 
+  patches = [
+    (fetchpatch2 {
+      # open PR: https://github.com/mozilla/cbindgen/pull/1010
+      # see also: https://github.com/NixOS/nixpkgs/pull/298108
+      name = "1010-fix-test-failures-due-to-CARGO_BUILD_TARGET";
+      url = "https://patch-diff.githubusercontent.com/raw/mozilla/cbindgen/pull/1010.patch";
```

Comment:
this needs to be a stable url, like: `https://github.com/mozilla/cbindgen/commit/f8efe01d7f23c56b593ec686f0b430e6d47404e0.patch?full_index=1`, and the other commit, but also needs to be merged in the tree otherwise the commits can get garbage collected -- so with unmerged PRs we are just vendoring the patch rather than using `fetchpatch`.

Reply:
Good idea! I will wait till next week and see if it can get merged upstream. They are pretty responsive.

> stable url, like: `https://github.com/mozilla/cbindgen/commit/f8efe01d7f23c56b593ec686f0b430e6d47404e0.patch?full_index=1`

I have heard this before and will do so once upstream has responded. But just out of curiosity: how might `https://patch-diff.githubusercontent.com/raw/*/*/pull/*.patch` differ from the stable URL? Maybe the hash will change?

Reply:
` https://github.com/mozilla/cbindgen/pull/1010.patch` will change when the PR is updated, thus creating hash mismatch and breaking the package.